packages/xo-web/src/xo-app/backup-ng/new/metadata/index.js

```diff
@@ -0,0 +1,419 @@
+        _pools,
+      }),
+      setSchedules: (_, _schedules) => () => ({
+        _schedules,
+      }),
+      setSettings: (_, _settings) => () => ({
+        _settings,
+      }),
+      toggleMode: (_, { mode }) => state => ({
+        [`_${mode}`]: !state[mode],
+      }),
+      addRemote: (_, { id }) => state => ({
+        _remotes: [...state.remotes, id],
+      }),
+      deleteRemote: (_, key) => state => {
+        const _remotes = [...state.remotes]
+        _remotes.splice(key, 1)
+        return {
+          _remotes,
+        }
+      },
+    },
+    computed: {
+      idForm: generateId,
+
+      modePoolMetadata: ({ _modePoolMetadata }, { job }) =>
+        defined(_modePoolMetadata, () => !isEmpty(destructPattern(job.pools))),
+      modeXoMetadata: ({ _modeXoMetadata }, { job }) =>
+        defined(_modeXoMetadata, () => job.xoMetadata),
+      name: (state, { job }) => defined(state._name, () => job.name, ''),
+      pools: ({ _pools }, { job }) =>
+        defined(_pools, () => destructPattern(job.pools)),
+      retentions: ({ modePoolMetadata, modeXoMetadata }) => {
+        const retentions = []
+        if (modePoolMetadata) {
+          retentions.push(RETENTION_POOL_METADATA)
+        }
+        if (modeXoMetadata) {
+          retentions.push(RETENTION_XO_METADATA)
+        }
+        return retentions
+      },
+      schedules: ({ _schedules }, { schedules }) =>
+        defined(_schedules, schedules),
+      settings: ({ _settings }, { job }) =>
+        defined(_settings, () => job.settings),
+      remotes: ({ _remotes }, { job }) =>
+        defined(_remotes, () => destructPattern(job.remotes), []),
+      remotesPredicate: ({ remotes }) => ({ id }) => !remotes.includes(id),
+
+      isJobInvalid: state =>
+        state.missingModes ||
+        state.missingName ||
+        state.missingPools ||
+        state.missingRemotes ||
+        state.missingRetentionPoolMetadata ||
+        state.missingRetentionXoMetadata ||
+        state.missingSchedules,
+
+      missingModes: state => !state.modeXoMetadata && !state.modePoolMetadata,
+      missingName: state => state.name.trim() === '',
+      missingPools: state => state.modePoolMetadata && isEmpty(state.pools),
+      missingRemotes: state => isEmpty(state.remotes),
+      missingRetentionPoolMetadata: state =>
+        state.modePoolMetadata &&
+        !some(
+          state.settings,
+          ({ retentionPoolMetadata }) => retentionPoolMetadata !== null
+        ),
+      missingRetentionXoMetadata: state =>
+        state.modeXoMetadata &&
+        !some(
+          state.settings,
+          ({ retentionXoMetadata }) => retentionXoMetadata !== null
+        ),
+      missingSchedules: state => isEmpty(state.schedules),
+    },
+  }),
+  injectState,
+  ({ state, effects, job, remotes }) => {
+    const [submitHandler, submitTitle] =
+      job === undefined
+        ? [effects.createJob, 'formCreate']
+        : [effects.editJob, 'formSave']
```

Comment:
Don't you want to do this in a computed?

Reply:
We cannot access to the effects form the computed.